tests/wasmfs/wasmfs_seek.c

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2021 The Emscripten Authors.  All rights reserved.
+ * Emscripten is available under two separate licenses, the MIT license and the
+ * University of Illinois/NCSA Open Source License.  Both these licenses can be
+ * found in the LICENSE file.
+ */
+
+#include <assert.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <stdio.h>
+#include <unistd.h>
+
+// FIXME: The existing io.c file implements devices, which haven't been
+// implemented in the new FS yet.
+// This test simply implements the seek portion.
+
+/*
+ * Copyright 2021 The Emscripten Authors.  All rights reserved.
+ * Emscripten is available under two separate licenses, the MIT license and
+ * the University of Illinois/NCSA Open Source License.  Both these licenses
+ * can be found in the LICENSE file.
+ */
+
+#include <assert.h>
+#include <emscripten.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <stdio.h>
+#include <string.h>
+#include <sys/uio.h>
+#include <unistd.h>
+
+// FIXME: Merge this standalone test back into dup.c after new FS can support
+// it.
+
+int main() {
+
+  char readBuffer[256] = {0};
+  char writeBuffer[] = "writeme";
+
+  int f = open("/file", O_RDWR | O_CREAT);
+  const char* msg = "1234567890";
+  write(f, msg, strlen(msg));
+
+  printf("seek: %lld\n", lseek(f, 3, SEEK_SET));
+  printf("errno: %d\n\n", errno);
+  printf("partial read from file: %zd\n", read(f, readBuffer, 3));
+  printf("data: %s\n", readBuffer);
+  memset(readBuffer, 0, sizeof readBuffer);
+  printf("errno: %d\n\n", errno);
+  errno = 0;
+
+  printf("seek: %lld\n", lseek(f, -2, SEEK_END));
+  printf("errno: %d\n", errno);
+  errno = 0;
+  printf("partial read from end of file: %zd\n", read(f, readBuffer, 3));
+  printf("data: %s\n", readBuffer);
+  memset(readBuffer, 0, sizeof readBuffer);
+  printf("errno: %d\n\n", errno);
+  errno = 0;
+
+  printf("seek: %lld\n", lseek(f, -15, SEEK_CUR));
+  printf("errno: %d\n", errno);
+  errno = 0;
+  printf("partial read from before start of file: %zd\n",
+         read(f, readBuffer, 3));
+  printf("data: %s\n", readBuffer);
+  memset(readBuffer, 0, sizeof readBuffer);
+  printf("errno: %d\n\n", errno);
+  errno = 0;
+
+  printf("seek: %lld\n", lseek(f, 0, SEEK_SET));
+  printf("write to start of file: %zd\n", write(f, writeBuffer, 3));
+  printf("errno: %d\n\n", errno);
+  errno = 0;
+
+  printf("seek: %lld\n", lseek(f, 0, SEEK_END));
+  printf("write to end of file: %zd\n", write(f, writeBuffer, 3));
+  printf("errno: %d\n\n", errno);
+  errno = 0;
+
+  printf("seek: %lld\n", lseek(f, 10, SEEK_END));
+  printf("write after end of file: %zd\n",
+         write(f, writeBuffer, sizeof writeBuffer));
```

Comment:
It looks like this write did not actually write the data? I would expect to see "writeme" at the end of the file contents.

Reply:
Just fixed this: `wri4567890wri\0\0\0\0\0\0\0\0\0\0writeme\0`